src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,371 @@
+                                    RecommendationConfigItem>> recommendationsConfig = termRecommendations.getCostRecommendations().getConfig();
+
+                    Double cpuRecommendationValue = recommendationsConfig.get(AnalyzerConstants.ResourceSetting.requests).get(AnalyzerConstants.RecommendationItem.CPU).getAmount();
+                    Double memoryRecommendationValue = recommendationsConfig.get(AnalyzerConstants.ResourceSetting.requests).get(AnalyzerConstants.RecommendationItem.MEMORY).getAmount();
+
+                    LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.RECOMMENDATION_VALUE, "CPU", containerName, cpuRecommendationValue));
+                    LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.RECOMMENDATION_VALUE, "MEMORY", containerName, memoryRecommendationValue));
+
+                    String cpuRecommendationValueForVpa = resource2str("CPU", cpuRecommendationValue);
+                    String memoryRecommendationValueForVpa = resource2str("MEMORY", memoryRecommendationValue);
+
+                    // creating container resource vpa object
+                    RecommendedContainerResources recommendedContainerResources = new RecommendedContainerResources();
+                    recommendedContainerResources.setContainerName(containerName);
+
+                    // setting target values
+                    Map<String, Quantity> target = new HashMap<>();
+                    target.put("cpu", new Quantity(cpuRecommendationValueForVpa));
+                    target.put("memory", new Quantity(memoryRecommendationValueForVpa));
+
+                    // setting lower bound values
+                    Map<String, Quantity> lowerBound = new HashMap<>();
+                    lowerBound.put("cpu", new Quantity(cpuRecommendationValueForVpa));
+                    lowerBound.put("memory", new Quantity(memoryRecommendationValueForVpa));
+
+                    // setting upper bound values
+                    Map<String, Quantity> upperBound = new HashMap<>();
+                    upperBound.put("cpu", new Quantity(cpuRecommendationValueForVpa));
+                    upperBound.put("memory", new Quantity(memoryRecommendationValueForVpa));
+
+                    recommendedContainerResources.setLowerBound(lowerBound);
+                    recommendedContainerResources.setTarget(target);
+                    recommendedContainerResources.setUpperBound(upperBound);
+
+                    containerRecommendations.add(recommendedContainerResources);
+                }
+            } else {
+                LOGGER.error(AnalyzerErrorConstants.RecommendationUpdaterErrors.RECOMMENDATION_DATA_NOT_PRESENT);
+            }
+        }
+        return containerRecommendations;
+    }
+
+    /**
+     * This function converts the cpu and memory values to VPA desired format
+     */
+    public static String resource2str(String resource, double value) {
```

Comment:
I feel this function can be widely used across project, Can we move it to an utility class where other services also can make use of it in future?

Reply:
Agree. I have moved this function to `RecommendationUtils` under `analyzer.recommendation.utils` package.